log-manager/src/main/java/io/airlift/log/Logging.java

```diff
@@ -81,15 +81,22 @@ public static synchronized Logging initialize()
 
     private Logging()
     {
-        ROOT.setLevel(Level.INFO.toJulLevel());
+        resetConfiguration();
+        log.info("Logging to stderr");
+
+        redirectStdStreams();
+    }
+
+    private void resetConfiguration()
+    {
+        // Reset named loggers, remove their handlers, and reset root to INFO
+        LogManager.getLogManager().reset();
```

Comment:
It looks like this will drop logs when it does the reset if the appenders haven't flushed yet. You might be safer doing a no-op after the initial configuration if this is called a second time.

Reply:
Hmm, good point. There are two issues:
* Appenders may not implement close() correctly by flushing their buffers.
* The app may log something after the appenders were removed, and before the new appenders are added.
I'll close this PR.